refactor: updated xo dependency

mfranzke opened the pull request:
## Proposed changes

<!-- Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue. -->

## Types of changes

<!-- What types of changes does your code introduce?
_Put an `x` in the boxes that apply_ -->

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [x] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

<!-- ## Checklist

_Put an `x` in the boxes that apply.

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
-->

## Further comments

<!-- If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

❤️ Thank you!
-->

annsch commented:
@mfranzke could you explain why you switched from replace to replaceAll? This could lead from an issue in XO but for review issues it would be easier to understand if it's documented in PR's description ... without explanation I don't understand the mix of this PR :)

mfranzke commented:
> @mfranzke could you explain why you switched from replace to replaceAll? This could lead from an issue in XO but for review issues it would be easier to understand if it's documented in PR's description ... without explanation I don't understand the mix of this PR :)

valid aspect, thanks. I've added `xo` code styles feedback as well to this PR, and added this aspects within the description.

mfranzke commented:
we're still facing an error due to https://github.com/typescript-eslint/typescript-eslint/issues/8276 (which get mentioned within https://github.com/typescript-eslint/typescript-eslint/pull/8262 to probably solve the problem described in there).

mfranzke commented on .xo-config.cjs:
```diff
@@ -56,6 +56,7 @@ module.exports = {
 			}
-		]
+		],
+		'unicorn/prevent-abbreviations': 0
```
created https://github.com/db-ui/mono/issues/2255 as a follow up.

nmerget commented on packages/foundations/scripts/zeplin-styleguide.js:
```diff
@@ -53,9 +53,7 @@ const mergeData = (data) => {
-				if (!temporaryData[sKey]) {
-					temporaryData[sKey] = {};
-				}
+				temporaryData[sKey] ||= {};
```
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR_assignment 

okay :O
